src/control/server/interceptors.go

```diff
@@ -145,15 +145,19 @@ func checkVersion(ctx context.Context, self *build.VersionedComponent, req inter
 	}
 
 	other, err := build.NewVersionedComponent(buildComponent, otherVersion)
-	if err != nil || other.Version.IsZero() {
+	if err != nil {
 		other = &build.VersionedComponent{
 			Component: "unknown",
 			Version:   build.MustNewVersion(otherVersion),
 		}
 		return FaultIncompatibleComponents(self, other)
 	}
 
-	if err := build.CheckCompatibility(self, other); err != nil {
+	// Set per-release compatibility rules here.
+	releaseCompatRules := []*build.InteropRule{
+		build.Server22xAgent20x,
+	}
```

Comment:
Why wouldn't we want to lump this in with the general rules? If the intent is to make it go away for later releases, I don't think it's easier to find here.

Reply:
The idea (which I will admit may be misguided) is that the general rules are meant to be generally stable across releases and therefore not the place to look for understanding release-specific requirements. In this case, we have a requirement on the 2.2 branch that we treat an unversioned agent as a 2.0.x agent and allow it to access a 2.2 server. That rule will not exist on other branches.